zipline/utils/cache.py

```diff
@@ -86,6 +88,33 @@ def _unsafe_get_value(self):
         return self._value
 
 
+def basic_cleanup(cache, key):
+    del cache[key]
+
+
+def dataframe_gc_cleanup(cache, key):
+    # Try to deterministically garbage collect the previous result by
+    # removing any references to it. There are at least three sources
+    # of references:
+
+    # 1. self._pipeline_cache holds a reference.
```

Comment:
This comment is referencing variables from TradingAlgorithm, which doesn't really make sense if this is a standalone function.